### PR TITLE
[test_system_health] Align system health test cases with production code and fix issues

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:

1. Aligh system health test cases with latest production code
2. Fix issue when mocking fan speed value
3. Combine multiple mock together to save test time

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Fix test cases to align with latest production code

#### How did you do it?

1. Aligh system health test cases with latest production code
2. Fix issue when mocking fan speed value
3. Combine multiple mock together to save test time

#### How did you verify/test it?

Manually run the test cases

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

File: tests/system_health/test_system_health.py [hunk 1]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
-                                                                                                  THERMAL_CHECK_INTERVAL))
-            time.sleep(PSU_CHECK_INTERVAL)
-            value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, psu_name)
-            assert value and expect_value in value, 'Mock PSU hot, expect {}, but got {}'.format(expect_value,
-                                                                                                 value)
+        fan_mock_result, fan_name = device_mocker.mock_fan_status(True)
+        psu_mock_result, psu_name = device_mocker.mock_psu_temperature(True)
+        if fan_mock_result and psu_mock_result:
+            logging.info('Mocked fan good for {}'.format(fan_name))
+            logging.info('Mocked PSU cold for {}'.format(psu_name))
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:202]: “cold” typo?

File: tests/system_health/test_system_health.py [hunk 2]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
+            assert not value or value != fan_expect_value, 'Mock fan normal, but it still report broken'
 
-        mock_result, psu_name = device_mocker.mock_psu_temperature(True)
-        if mock_result:
-            logging.info('Mocked PSU cold for {}, waiting {} seconds for it to take effect'.format(psu_name,
-                                                                                                   THERMAL_CHECK_INTERVAL))
-            time.sleep(PSU_CHECK_INTERVAL)
             value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, psu_name)
-            assert not value or expect_value not in value, 'Mock PSU cold, but it is still hot'
+            assert not value or psu_expect_value not in value, 'Mock PSU cold, but it is still hot'
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:209]: cold?

File: tests/system_health/test_system_health.py [hunk 3]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
-                                                                                                      value)
+            assert not value or psu_expect_value != value, 'Mock PSU power good, but it is still out of power'
+
+        fan_mock_result, fan_name = device_mocker.mock_fan_status(False)
+        fan_expect_value = EXPECT_FAN_BROKEN.format(fan_name)
+        psu_mock_result, psu_name = device_mocker.mock_psu_temperature(False)
+        psu_expect_value = EXPECT_PSU_HOT.format(psu_name)
+        if fan_mock_result and psu_mock_result:
+            logging.info('Mocked fan broken for {}'.format(fan_name))
+            logging.info('Mocked PSU hot for {}'.format(psu_name))
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:187]: hot --> overheated?

File: tests/system_health/test_system_health.py [hunk 4]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
             value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, psu_name)
-            assert not value or expect_value != value, 'Mock PSU presence, but it is still absence'
+            assert value and psu_expect_value == value, 'Mock PSU no power, expect {}, but got {}'.format(psu_expect_value,
+                                                                                                          value)
+
+        fan_mock_result, fan_name = device_mocker.mock_fan_presence(True)
+        psu_mock_result, psu_name = device_mocker.mock_psu_status(True)
+        if fan_mock_result and psu_mock_result:
+            logging.info('Mocked fan presence for {}'.format(fan_name ))
+            logging.info('Mocked PSU good power for {}'.format(psu_name, THERMAL_CHECK_INTERVAL))
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:170]: two parameters?

File: tests/system_health/test_system_health.py [hunk 5]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
+        asic_mock_result = device_mocker.mock_asic_temperature(True)
+        psu_mock_result, psu_name = device_mocker.mock_psu_presence(True)
+        if fan_mock_result and asic_mock_result and psu_mock_result:
+            logging.info('Mocked valid fan speed for {}'.format(fan_name))
+            logging.info('Mocked ASIC cold')
+            logging.info('Mocked PSU presence for {}'.format(psu_name))
+            logging.info('Waiting {} seconds for it to take effect'.format(THERMAL_CHECK_INTERVAL))
             time.sleep(THERMAL_CHECK_INTERVAL)
+            value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, fan_name)
+            assert not value or fan_expect_value not in value, 'Mock fan valid speed, expect {}, ' \
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:140]: missing format?

File: tests/system_health/test_system_health.py [hunk 6]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
-            logging.info('Mocked ASIC cold, waiting {} seconds for it to take effect'.format(THERMAL_CHECK_INTERVAL))
+            value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, psu_name)
+            assert value and psu_expect_value == value, 'Mock PSU absence, expect {}, but got {}'.format(psu_expect_value,
+                                                                                                         value)
+        fan_mock_result, fan_name = device_mocker.mock_fan_speed(True)
+        asic_mock_result = device_mocker.mock_asic_temperature(True)
+        psu_mock_result, psu_name = device_mocker.mock_psu_presence(True)
+        if fan_mock_result and asic_mock_result and psu_mock_result:
+            logging.info('Mocked valid fan speed for {}'.format(fan_name))
+            logging.info('Mocked ASIC cold')
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:135]: the log is ambiguous.

File: tests/system_health/test_system_health.py [hunk 7]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
-            logging.info('Mocked ASIC hot, waiting {} seconds for it to take effect'.format(THERMAL_CHECK_INTERVAL))
-            time.sleep(THERMAL_CHECK_INTERVAL)
+            assert value and fan_expect_value in value, 'Mock fan invalid speed, expect {}, but got {}'.format(fan_expect_value,
+                                                                                                               value)
             value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, 'ASIC')
-            assert value and expect_value in value, 'Mock ASIC temperature hot, expect {}, but got {}'.format(
-                expect_value,
+            assert value and asic_expect_value in value, 'Mock ASIC temperature hot, expect {}, but got {}'.format(
+                asic_expect_value,
                 value)
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:125]: move to same line

Comment (keboliu) [inline tests/system_health/test_system_health.py:125]: move to above line?

File: tests/system_health/test_system_health.py [hunk 8]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
             value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, fan_name)
-            assert not value or value != expect_value, 'Mock fan normal, but it still report broken'
-
-        mock_result = device_mocker.mock_asic_temperature(False)
-        expect_value = EXPECT_ASIC_HOT
-        if mock_result:
-            logging.info('Mocked ASIC hot, waiting {} seconds for it to take effect'.format(THERMAL_CHECK_INTERVAL))
-            time.sleep(THERMAL_CHECK_INTERVAL)
+            assert value and fan_expect_value in value, 'Mock fan invalid speed, expect {}, but got {}'.format(fan_expect_value,
+                                                                                                               value)
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:121]: some formatting issue?

File: tests/system_health/test_system_health.py [hunk 9]
```diff
@@ -100,129 +101,112 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
-            value = redis_get_field_value(duthost, STATE_DB, HEALTH_TABLE_NAME, fan_name)
-            assert value and value == expect_value, 'Mock fan broken, expect {}, but got {}'.format(expect_value,
-                                                                                                    value)
-        mock_result, fan_name = device_mocker.mock_fan_status(True)
-        if mock_result:
-            logging.info('Mocked fan good for {}, waiting {} seconds for it to take effect'.format(fan_name,
-                                                                                                   THERMAL_CHECK_INTERVAL))
+        if fan_mock_result and asic_mock_result and psu_mock_result:
+            logging.info('Mocked invalid fan speed for {}'.format(fan_name))
+            logging.info('Mocked ASIC hot')
```
Comment (keboliu) [inline tests/system_health/test_system_health.py:115]: overheat?

Comment (keboliu) [inline tests/system_health/test_system_health.py:115]: overheat

**Review by keboliu — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
https://github.com/Azure/sonic-mgmt/pull/2898

**Pull request closed without merging**